Re-add make generate-sitemap action

nrichers opened the pull request:
# Pull Request Description

## What and why?
<!-- Briefly describe what changed and why — include behavior before and after if helpful -->
<!-- Frontend: Add screenshots or videos showing before/after -->

It appears https://github.com/validmind/documentation/commit/ae973d4600a1b99c0dbd94bf718fe72af37a2300#diff-97e4422bf78b64c199ffc642cb458be76ad2bd30b54a40c85846cfc7f2d35cebL63 removed the action to generate our sitemaps. Why, I'm not sure, but this PR fixes the issue. 

Without the sitemaps, the Lighthouse check will fail every time, as they have a dependency on there being a sitemap we can parse to check files. 

Relates to [sc-11350](https://app.shortcut.com/validmind/story/11350/fix-lighthouse-checks)

## How to test
<!-- Describe how the change has been tested, and how a reviewer can test the change locally -->

Proof's in the pudding: If the Lighthouse check passes, then this means we're getting the sitemaps. 

## What needs special review?
<!-- Optional: Call out specific areas needing detailed review -->

## Dependencies, breaking changes, and deployment notes
<!-- A list of links to pull requests that are depend on or are dependencies of this PR -->
<!-- Any special deployment considerations? -->
<!-- List any breaking changes -->

## Release notes
<!--- REPLACE THIS COMMENT WITH YOUR DESCRIPTION --->

<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `bug`
- `breaking-change`
- `deprecation`
- `documentation`
- `environment-variables`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## Checklist
- [x] What and why
- [ ] Screenshots or videos (Frontend)
- [x] How to test
- [ ] What needs special review
- [ ] Dependencies, breaking changes, and deployment notes
- [x] Labels applied
- [x] PR linked to Shortcut
- [ ] Unit tests added (Backend)
- [ ] Tested locally
- [ ] Documentation updated (if required)
- [ ] Environment variable additions/changes documented (if required)

validbeck reviewed (approved):
Oops, my turn to apologise — I removed that to test when the issues were happening with the notebook executions and never added them back in. Thanks for catching that!

nrichers commented:
> Oops, my turn to apologise — I removed that to test when the issues were happening with the notebook executions and never added them back in. Thanks for catching that!

No problem at all! The issue is now fixed, see https://github.com/validmind/documentation/actions/runs/16380964565/job/46292181470?pr=849 for a successful run:

<img width="450" height="579" alt="Capto_ 2025-07-18_03-17-21_pm" src="https://github.com/user-attachments/assets/162ca1b4-89b5-41b0-b0ef-c562c57dc305" />

That said, I have another change brewing that will improve usability. For context: 

- We currently check the PR preview site whenever we commit.
- After a commit, a new preview site won't be available right away, it takes time for the validate workflow to complete.
- This delay means we tend to check a preview site from a previous commit.

A much better way would be to trigger the Lighthouse check when the validate workflow has completed, meaning we always check the latest preview. Working on that now ...

nrichers commented:
Yup, that seems to work, merging: https://github.com/validmind/documentation/actions/runs/16381813365/job/46294681784?pr=849
